plugins/fs/src/scope.rs

```diff
@@ -56,8 +56,9 @@ impl Scope {
 
         {
             let mut allowed = self.allowed.lock().unwrap();
-            allowed.push(path.to_path_buf());
-            allowed.push(path.join(if recursive { "**" } else { "*" }));
+            let p = path.to_string_lossy();
+            allowed.push(escape(&p));
+            allowed.push(PathBuf::from(if recursive { "**" } else { "*" }));
```

Comment:
shouldn't this be `escape(&p).join(if recursive { "**" } else { "*" })`, otherwise I think we are allowing access to all and any directory